### ISSUE TITLE
Make React Tutorial use Hooks

### ISSUE THREAD

**Issue by aromalanil** [freeCodeCamp/freeCodeCamp#39711]
# Description
At present, the React section in Front End Library Certification is using Class Components. 
But since hooks are the new way of doing React and more people and adapting to Hooks, I suggest including Hooks in the program. 

Also, I think for beginners hooks will be more welcoming and easy to the React world.

**Comment by JimmyJoo** [comment]
Yes, I was just thinking the same. I would love to contribute to including Hooks.

**Comment by ojeytonwilliams** [comment]
@RandellDawson is the best person to guide here.  My understanding is that we'd love to include lessons with modern React concepts, but that we will be adding them in the upcoming curriculum, rather than the current one.

**Comment by aromalanil** [comment]
If the team is planning to add this to the upcoming curriculum, I would suggest doing that as soon as possible, as the frameworks are regularly being updated the new curriculum also may get irrelevant soon. Also as @JimmyJoo said I also would love to contribute to include hooks.

**Comment by ludocourbin** [comment]
Hello there ! I'm a  junior developer ( I've learned a lot from you website and would love to give back ) 
 I'm confident I can learn a lot by making the change from class component to hooks in your react tutorial, how can I get going ? ( first time collaborating ! 🎊 )

**Comment by RandellDawson** [comment]
> Hello there ! I'm a junior developer ( I've learned a lot from you website and would love to give back )
> I'm confident I can learn a lot by making the change from class component to hooks in your react tutorial, how can I get going ? ( first time collaborating !  )

@ludocourbin We are going to leave the existing class component lessons as there are still many code bases out there that use them still.  We will be incorporating new lessons in our project-based curriculum that is currently being built by staff.  We might post some GitHub issues around this in the future, but for now, you can take a look at any outstanding issue that has a `help wanted` label.  Looking forward to your contributions!

**Comment by ludocourbin** [comment]
Hello @RandellDawson, I'll go on and explore the issue with the corresponding tag as you mentioned. Thank you for your answer.

**Comment by Twaha-Rahman** [comment]
@ludocourbin If it is your first time contributing, you'll find the [contribution docs](https://contribute.freecodecamp.org/) very useful. Happy coding!

**Comment by ludocourbin** [comment]
Thank you @Twaha-Rahman for you helpful recommendation 😁

**Comment by shootermv** [comment]
as far as i can see - react version need to be upgraded first (16.4 has no support for hooks)

**Comment by moT01** [comment]
Thanks for the input, we will not be adjusting the existing challenges this much anymore in favor of the project based curriculum in the works. Thanks and happy coding 🎉
